### PR TITLE
Drop support of symfony 4

### PR DESCRIPTION
**Pull request by Chris53897**
Blocked by
- [x] matthiasnoback/symfony-dependency-injection-testsing 
- [x] https://github.com/php-http/client-common/pull/232

please see
https://github.com/meilisearch/meilisearch-symfony/pull/265#issuecomment-1788661930
and https://github.com/meilisearch/meilisearch-symfony/issues/300

### CODE REVIEW

**Review by curquiza — COMMENTED** [review]
Hello @Chris53897 

Don't forget to update the bors file accordingly, if you add or remove jobs in the CI 😊 

https://github.com/meilisearch/meilisearch-symfony/blob/main/bors.toml

**Comment by Chris53897** [comment]
Thanks for the reminder. I forgot it again :(
I have added it now.

**Comment by Chris53897** [comment]
Testruns for PHP 8.3 are blocked by https://github.com/shivammathur/setup-php/issues/789

**Review by 94noni — APPROVED** [review]

**Comment by norkunas** [comment]
This is missing Symfony 6.4

**Comment by Chris53897** [comment]
I have added tests for Symfony 6.4 now.
Now waiting for the blocking PR. (see on top of page)

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -52,10 +54,15 @@ jobs:
           tools: composer:v2, flex
       - name: Validate composer.json and composer.lock
         run: composer validate
+      - name: "Stability dev (temp until a stable release of symfony 7 is available)"
+        if: matrix.sf-version == '7.0.*'
+        run: |
+           composer config minimum-stability "dev"
+           composer config prefer-stable false
```
Comment (norkunas) [inline .github/workflows/tests.yml:61]: this is not needed anymore

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -77,6 +84,12 @@ jobs:
       - name: Validate composer.json and composer.lock
         run: composer validate
 
+      - name: "Stability dev (temp until a stable release of symfony 7 is available)"
+        if: matrix.sf-version == '7.0.*'
+        run: |
+           composer config minimum-stability "dev"
+           composer config prefer-stable false
+
```
Comment (norkunas) [inline .github/workflows/tests.yml:92]: this is not needed anymore

**Comment by norkunas** [comment]
Instead of [Empty-Commit](https://github.com/meilisearch/meilisearch-symfony/pull/301/commits/322bcc89c5dac48816114e562d37853d72446fbd) just do `git commit --amend --no-edit` :slightly_smiling_face:

**Comment by Chris53897** [comment]
Ok, will do. I saved it to my snippets.

**Comment by norkunas** [comment]
I'd conditionally require `var-exporter` when it is needed, like when we conditionally remove doctrine/annotations

**Comment by norkunas** [comment]
As we all want to release Symfony 7 support faster, then we should currently avoid dropping 4.4 support as two blocker issues popped up that needs to be fixed before

File: tests/Kernel.php [hunk 3]
```diff
@@ -40,5 +40,14 @@ protected function configureContainer(ContainerBuilder $container, LoaderInterfa
                 ],
             ]);
         }
+
+        // symfony/var-exporter >= 6.2
```
[the remark this answers is not in the document]
Comment (norkunas) [reply tests/Kernel.php:44]: I've included CI fix in #308

[the remark this answers is not in the document]
Comment (Chris53897) [reply tests/Kernel.php]: thanks @norkunas

**Comment by Chris53897** [comment]
@norkunas I added a new PR. But i am on holidays starting from friday (until 19.12). And not much time tomorrow.
Feel free to finish it.

**Review by norkunas — APPROVED** [review]

**Comment by norkunas** [comment]
bors merge

**Pull request merged**